src/CP/core/search/ilds.jl

```diff
@@ -0,0 +1,77 @@
+
+
+
+"""
+initroot!(toCall::Stack{Function}, ::Type{DFSearch},model::CPModel, variableHeuristic::AbstractVariableSelection, valueSelection::ValueSelection, newConstraints=nothing)
+
+generic function to instantiate the research based on a specific Strategy <: SearchStrategy. The max discrepancy correspond to the number or branchable variables at 
+the beginning of the search. Calls to expandIlds! with a decreasing discrepancy is stacked in the toCall Stack. 
+"""
+function initroot!(toCall::Stack{Function}, ::Type{ILDSearch}, model::CPModel, variableHeuristic::AbstractVariableSelection, valueSelection::ValueSelection, newConstraints=nothing)
+    isboundedlist = [!isbound(v) for (k,v) in model.variables]
+    @assert !isempty(isboundedlist) "initialisation failed : no declared variables"
+    depth = sum(isboundedlist)
+    
+    # Note that toCall stack is a LIFO data structure, expandIlds with a discrepancy threshold of 0 will be the first one to execute (then with 1, 2, 3, etc.)
+    for k in depth:-1:1
+        push!(toCall, (model) -> (nothing;expandIlds!(toCall,k,depth, nothing, model, variableHeuristic, valueSelection)))
+    end
```

Comment:
I'm not sure to see why there is `nothing;` in the function

Reply:
For generalization purposes, functions pushed in the stack needed to have the same formalism as the one stacked during the DFS process.